src/device/gcn/hostcall.jl

```diff
@@ -370,9 +364,9 @@ function hostcall_host_wait(signal_handle::HSA.Signal; maxlat=DEFAULT_HOSTCALL_L
     @debug "Hostcall: Waiting on signal $signal_handle"
     start_time = time_ns()
     while !Runtime.RT_EXITING[]
-        prev = get_value(signal_handle)
+        prev = host_signal_load(signal_handle)
         if prev == DEVICE_MSG_SENTINEL
-            prev = set_value_acq_rel!(
+            prev = cmpxchg_acq_rel!(
```

Comment:
Maybe this can be `host_signal_cmpxchg!`? Sorry for the bad original suggestion...